modules/db/src/main/scala/gem/dao/ObservationDao.scala

```diff
@@ -11,23 +11,39 @@ import scalaz._, Scalaz._
 object ObservationDao {
 
   def insert(o: Observation[StaticConfig, _]): ConnectionIO[Int] =
-    Statements.insert(o).run
+    for {
+      id <- StaticConfigDao.insert(o.staticConfig)
```

Comment:
It will be run in a transaction yes.  In the `Importer` for example all the work associated with importing a single program or observation is done in a transaction by calling `transact` on the `ConnectionIO` we end up with.